src/services/mongodb/collection.js

```diff
@@ -143,109 +143,56 @@ Collection.prototype.upsert =
 
 // private
 function insertOp(self, docs, options) {
-  let stages = [];
-
-  // there may be no docs, when building for chained pipeline stages in
-  // which case the source is considered to be the previous stage
-  if (docs) {
```

Comment:
[ ⚠️  ]: no longer leaving the potential for no `docs` to be provided to this method, as chaining stages is no longer supported